controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1707,8 +1707,37 @@ func (r *BareMetalHostReconciler) updateEventHandler(e event.UpdateEvent) bool {
 }
 
 // SetupWithManager registers the reconciler to be run by the manager
-func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager, preprovImgEnable bool) error {
+func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager, preprovImgEnable bool, options controller.Options) error {
 
+	maxConcurrentReconciles := 0
+	if options.MaxConcurrentReconciles > 0 {
+		ctrl.Log.Info(fmt.Sprintf("Controller Concurrency will be set to %d according to command line flag", options.MaxConcurrentReconciles))
+	} else {
+		mCR, err := getMaxConcurrentReconciles()
+		if err != nil {
+			return err
+		}
+		maxConcurrentReconciles = mCR
+	}
+	options.MaxConcurrentReconciles = maxConcurrentReconciles
```

Comment:
This always sets it to 0 AFAICT.

Could we have single implementation instead of having to fix this in 5 different places? Like move everything inside getMaxConcurrentReconciles maybe.

Reply:
not sure if I got your point

Reply:
Say the user uses the CLI flag to set `options.MaxConcurrentReconciles` to, say, 15. Then:

```
	maxConcurrentReconciles := 0
```
This will be important later.
```
	if options.MaxConcurrentReconciles > 0 {
		ctrl.Log.Info(fmt.Sprintf("Controller Concurrency will be set to %d according to command line flag", options.MaxConcurrentReconciles))
```
Prints a log message saying we will use 15.
```
	} else {
		mCR, err := getMaxConcurrentReconciles()
		if err != nil {
			return err
		}
		maxConcurrentReconciles = mCR
	}
```
The road not taken.
```
	options.MaxConcurrentReconciles = maxConcurrentReconciles
```
Unconditionally sets `options.MaxConcurrentReconciles` to 0 regardless of the value passed in, then passes the options struct to the manager.

Reply:
ow snap, yes I had the assignment line which I misplaced during refactoring, good catch , fixed now.

Reply:
It's fixed here (albeit in a slightly convoluted way), but not in the other 3 copies.
This is why I suggested moving this to a single implementation.